tests/event_queue.cpp

```diff
@@ -0,0 +1,164 @@
+
+  SUBCASE("Custom threshold size should be used instead of the default one") {
+    countly.setEventsToRQThreshold(90); // before start
+    test_utils::initCountlyWithFakeNetworking(true, countly);
+
+    // generate 100 events
+    test_utils::generateEvents(100, countly);
+
+    // new threshold size is 90 so we should have 10 events in the EQ left
+    CHECK(countly.checkPersistentEQSize() == 10);
+
+    // RQ should have the 90 events
+    test_utils::checkTopRequestEventSize(90, countly);
+  }
+
+  // Setting the threshold size after start
+  SUBCASE("Custom threshold size should be used instead of the default one") {
+    test_utils::initCountlyWithFakeNetworking(true, countly);
+    countly.setEventsToRQThreshold(90); // after start
+
+    // generate 100 events
+    test_utils::generateEvents(100, countly);
+
+    // new threshold size is 90 so we should have 10 events in the EQ left
+    CHECK(countly.checkPersistentEQSize() == 10);
+
+    // RQ should have the 90 events
+    test_utils::checkTopRequestEventSize(90, countly);
+  }
+
+  // Setting a negative threshold size
+  SUBCASE("UINT_MAX should be used instead of the default one") {
+    countly.setEventsToRQThreshold(-6); // before start
+    test_utils::initCountlyWithFakeNetworking(true, countly);
+    // generate 10000 events
+    test_utils::generateEvents(1, countly);
+
+    // threshold is now set to UINT_MAX (as we use unsigned int) so we should have all events still in the EQ
+    CHECK(countly.checkPersistentEQSize() == 0);
+
+    test_utils::checkTopRequestEventSize(1, countly);
+  }
+
+  // Setting threshold size both before and after start
+  SUBCASE("custom threshold should be used instead of default threshold") {
+    countly.setEventsToRQThreshold(-5); // before start
+    test_utils::initCountlyWithFakeNetworking(true, countly);
+    countly.setEventsToRQThreshold(90); // after start
+
+    // generate 100 events
+    test_utils::generateEvents(100, countly);
+
+    // new threshold size is 90 so we should have 10 events in the EQ left
+    CHECK(countly.checkPersistentEQSize() == 10);
+
+    // RQ should have the 90 events
+    test_utils::checkTopRequestEventSize(90, countly);
+  }
+
+  // Setting threshold size both before and after start, with update session in between
+  SUBCASE("custom threshold should be used instead of default threshold") {
+    countly.setEventsToRQThreshold(-5); // before start
+    test_utils::initCountlyWithFakeNetworking(true, countly);
+    countly.updateSession();
+    countly.setEventsToRQThreshold(90); // after start
+
+    // generate 100 events
+    test_utils::generateEvents(100, countly);
+
+    // new threshold size is 90 so we should have 10 events in the EQ left
+    CHECK(countly.checkPersistentEQSize() == 10);
+
+    // RQ should have the 90 events
+    test_utils::checkTopRequestEventSize(90, countly);
+  }
+
+  // Setting threshold size both before and after start, non-merge device ID change
```

Comment:
missing the "why" and potentially the "expected" result. what are we checking for?